### PR TITLE
Add stories for o-icons demos

### PR DESCRIPTION
**Pull request by ivomurrell**
All the different icons are selectable from a drop-down menu instead of all being rendered at once. The list of options is derived from the `fticons` imageset.

Custom icons are a little weird as they're just a span calling with a CSS class set, and the real work happens in the Sass. It doesn't seem worth creating a TSX component for that. It's also a bit harder to see what's going on under the hood compared to other components as we surface the generated HTML in a Storybook addon, but not the CSS. You can still easily inspect the code in your browser though, of course.

### CODE REVIEW

File: components/o-icons/src/tsx/icon.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface IconProps {
+	// currently no way to get a const type from a JSON import in TypeScript so
+	// let's stick with a string type for now
+	icon: string;
+}
```
Comment (chee) [inline components/o-icons/src/tsx/icon.tsx:5]: not for this PR but we could generate this `interface` as part of a build step

File: components/o-icons/src/tsx/icon.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {IconName} from './data';
+
+export interface IconProps {
+	// currently no way to get a const type from a JSON import in TypeScript so
+	// let's stick with a string type for now
+	icon: IconName;
```
Comment (chee) [inline components/o-icons/src/tsx/icon.tsx:6]:
```suggestion
	icon: IconName;
```

File: components/o-icons/src/scss/_icon-list.scss [hunk 3]
```diff
@@ -64,7 +64,5 @@ $o-icons-list: (
 	video,
 	warning-alt,
 	warning,
-	written-text,
-	eye-open,
-	eye-closed
+	written-text
```
[the remark this answers is not in the document]
Comment (chee) [reply components/o-icons/src/scss/_icon-list.scss:67]: 👁️👀👁️

**Comment by notlee** [comment]
Just done a little rebase there.

**Review by notlee — CHANGES_REQUESTED** [review]
Sorry almost merged but I see chee noticed something happened to the eye icons. When they're restored I think this is good to go :) 

https://github.com/Financial-Times/origami/pull/732/files#r874880314

File: components/o-icons/stories/icons.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+@import "@financial-times/o-icons/main";
+@include oIcons();
+
+.demo-icon-custom {
+	@include oIconsContent('lock');
+}
+
+.demo-icon-custom-color {
```
Comment (notlee) [inline components/o-icons/stories/icons.scss:8]: Please delete the custom colour story! Sorry I didn't think to mention. We decided to only show demos in storybook that are actually approved by design and part of the FT design system, which means we don't show icons with random "this is possible" colours

File: components/o-icons/src/scss/_icon-list.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (ivomurrell) [reply components/o-icons/src/scss/_icon-list.scss]: This script uses the imageset from https://github.com/Financial-Times/fticons (which supposedly replaces `o-icons`...? huh) which doesn't include the newly added 👁️ icons. Maybe we should move away from that data source for the script in the future?

File: components/o-icons/scripts/build-icon-list.cjs [hunk 5]
```diff
@@ -3,16 +3,19 @@
...
 
 function runScript() {
-	let {images} = loadJson(`${__dirname}/../node_modules/@financial-times/fticons/imageset.json`);
+	let {images} = loadJson(
+		`${__dirname}/../../../node_modules/@financial-times/fticons/imageset.json`
+	);
 	images = images.filter(image => !iconBlacklist.includes(image.name));
 	writeDemoData(images, `${__dirname}/../demos/src/data.json`);
 	writeSassIconList(images, `${__dirname}/../src/scss/_icon-list.scss`);
+	writeTsInterface(images, `${__dirname}/../src/tsx/data.ts`);
```
Comment (notlee) [inline components/o-icons/scripts/build-icon-list.cjs:18]: 🙌

**Pull request merged**